llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azurecosmosmongovcore/llama_index/storage/kvstore/azurecosmosmongovcore/base.py

```diff
@@ -0,0 +1,160 @@
+import logging
+from abc import ABC
+from typing import Dict, Optional
+
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.storage.kvstore.types import BaseKVStore, DEFAULT_COLLECTION
+from pymongo import MongoClient
+
+logger = logging.getLogger(__name__)
+
+APP_NAME = "Llama-Index-CDBMongoVCore-KVStore-Python"
+
+
+class AzureCosmosMongoVCoreKVStore(BaseKVStore, ABC):
+    """Creates an AzureCosmosMongoVCoreKVStore."""
```

Comment:
Is there a reason that the existing mongo kvstore doesn't work for azure here?

Reply:
Seems like its using pymongo under the hood, so I would expect the existing mongo kvstore (and other associated integrations) to work fine 👀

Reply:
@logan-markewich , we want to keep our integration separate with a different user agent for tracking purposes.

Reply:
@aayush3011 can't we just insert the user-agent in the existing mongodb integrations? Seems like something generally useful for anyone

Otherwise we have completely duplicate code just for a small difference, feels kind of icky

Reply:
@logan-markewich we would have to somehow figure out that this is an azure cosmos mongoDB and insert the necessary user agent. To do so, the user then needs to send some keyword like"azure" or "cosmos" as a property when creating using the kvStore.

Reply:
@logan-markewich And another thing can you please re open this [PR](https://github.com/run-llama/llama_index/pull/16138). It has the changes for Azure CosmosDB NoSql Storage Integrations. You closed the PR by mistake.

Reply:
@aayush3011 why can't the user agent be generic? `llama-index` or similar? You will know its using azure/cosmos because its connecting to those services

Reply:
The kvstore is only used for doctores and index stores -- still not seeing the argument for introducing all this code duplication and bloat just to add a user agent

We can have `llama-index-docstore-py`/`llama-index-kvstore-py`/`llama-index-index-store-py` be the user agent for the existing mongodb kvstores for examples

Reply:
Sure, I get your point. Would this be okay if we ask the customer to send "api=azure" in the kwargs and place a check for it to add the azure cosmos db specific user agent. If this approach doesn't seems good, we can go with what you suggested.

Reply:
Both seem fine to me, although the approach with the least amount of effort from the user will probably be the easiest

Reply:
Made the changes, please review.